File: recruit/src/main/java/com/arenella/recruit/beans/Candidate.java
package com.arenella.recruit.beans;

import java.time.LocalDate;

/**
* Class represents a Recruitment Candidate. A Candidate that 
* can be place on a project
* @author K Parkings
*/
public class Candidate {

	public static enum COUNTRY {NETHERLANDS, BELGIUM, UK}
	
	private String 		candidateId;
	private COUNTRY 	country;
	private String 		city;
	private boolean 	perm;
	private boolean 	freelance;
	private int			yearsExperience;
	private boolean 	available;
	private LocalDate 	registerd;
	private LocalDate 	lastAvailabilityCheck;
	
	/**
	* Returns the unique identifier of the Candidate
	* @return unique Id of the candidate
	*/
	public String getCandidateId() {
		return candidateId;
	}
	
	/**
	* Returns the Country in which the candidate is 
	* located
	* @return Country 
	*/
	public COUNTRY getCountry() {
		return country;
	}
	
	/**
	* Returns the City where the Candidate is located
	* @return Name of City where Candidate is located
	*/
	public String city() {
		return city;
	}
	
	/**
	* 
	* @return
	*/
	public boolean isFreelance() {
		return freelance;
	}
	
	/**
	* 
	* @return
	*/
	public boolean isPerm() {
		return perm;
	}
	
}

File: recruit-gui/src/app/accounts/candidate.ts

/**
* Represnets a candidate
* @author: K Parkings 
*/
export class Candidate {

	/**
	* Unique identifier of the Candidate
	*/
	candidateId:		string 			= '';
	
	/**
	* first name of the Candidate
	*/
	firstname:			string			= '';

	/**
	* Surname of the Candidate
	*/
	surname:			string 			= '';
	
	/**
	* Candidates email address
	*/
	email: 				string 			= '';
	
		/**
	* Country where the candidate is located 
	*/
	country: 			string 			= '';
	
	/**
	* City where the Candidate is located
	*/
	city: 				string 			= '';
	
	/**
	* Role sought by Candidates
	*/
	roleSought: 		string 			= '';
	
	/**
	* Whether the Candidates is aviable for work
	*/
	available:			boolean			= true;

}